### PR TITLE
Add all amqp.Publishing options for internal PublishOptions struct

### PR DESCRIPTION
**Pull request by hashbou**
This enables the possibility to fully utilise the given options for publisher by the amqp library.
Code is tested, formatted and linted with the given Makefile.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 name: Tests
 
 on:
+  push:
```
Comment (wagslane) [inline .github/workflows/tests.yml:4]: This should remain running on Pull requests, this ensures new PRs pass the tests

File: README.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ Wrapper of streadway/amqp that provides reconnection logic and sane defaults. Hi
 
 Supported by [Qvault](https://qvault.io)
 
-[![](https://godoc.org/github.com/wagslane/go-rabbitmq?status.svg)](https://godoc.org/github.com/wagslane/go-rabbitmq)![Deploy](https://github.com/wagslane/go-rabbitmq/workflows/Tests/badge.svg)
+[![](https://godoc.org/github.com/claranet/go-rabbitmq?status.svg)](https://godoc.org/github.com/claranet/go-rabbitmq)![Deploy](https://github.com/claranet/go-rabbitmq/workflows/Tests/badge.svg)
```
Comment (wagslane) [inline README.md:7]: Was this added by accident? We aren't changing the import path on the main repo

File: publish.go [hunk 3]
```diff
@@ -207,6 +292,15 @@ func (publisher *Publisher) Publish(
 		message.Body = data
 		message.Headers = tableToAMQPTable(options.Headers)
 		message.Expiration = options.Expiration
+		message.ContentEncoding = options.ContentEncoding
```
Comment (wagslane) [inline publish.go:295]: I really like that you're adding these options! Thanks.

Are we setting "sane" defaults for all of these new options though? Just want to double check your thoughts.

**Review by wagslane — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (hashbou) [reply .github/workflows/tests.yml:4]: Hi @wagslane Oh yes, I mixed things up while providing the PR but also make it available while the changes are not in your repository. I will clean this up and provide the PR again.

File: README.md (hunk 2 above)
Comment (hashbou) [reply README.md:7]: > Hi @wagslane Oh yes, I mixed things up while providing the PR but also make it available while the changes are not in your repository. I will clean this up and provide the PR again. 
Same here

**Comment by hashbou** [comment]
Will close this in favour of #50

**Pull request closed without merging**